app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt

```diff
@@ -46,6 +48,13 @@ class MainFragmentViewModel : ViewModel() {
     /** This is not an exact copy of the elements in the adapter  */
     var listElements = ArrayList<LayoutElementParcelable>()
 
+    var storageImages: ArrayList<LayoutElementParcelable>? = null
```

Comment:
Is it necessary to initialize everything to null and have to make it nullable?

Reply:
We're using null to find out if we've got anything saved in cache.

Reply:
Very probably can also be solved with some lazy initialization, but this way seems OK too.